accepted/00000-ansible-integration.md

```diff
@@ -0,0 +1,293 @@
+
+NOTE: The profile for a system with registered in Uyuni as "Foreign/ANSIBLE" does not look like as the one Salt/Traditional system, since most of the Uyuni features (channels, software profile, etc) are not implemented for neither "Foreign" nor "ANSIBLE" entitlements. In order to get the entire portfolio of Uyuni features for Salt minion, this "Foreign/ANSIBLE" systems would need to be transitioned to "Salt/ANSIBLE" minion.
+
+Notice also that the Ansible controller is not listed as systems list page, since there is not system entry for it, it would be shown in the respective new UI page for managing Ansible Controller.
+
+#### New UI tabs & pages
+
+* Ansible
+* Ansible / Ansible Controllers list
+* Ansible / Add new Ansible controller
+* Ansible / View Ansible controller information & and visualize playbooks (would also allow triggering playbooks via "salt-ssh" in the controller)
+
+## Operating Ansible
+
+This part describes different expectations/features to implement in order to execute certain operation in your Ansible infrastructure.
+
+#### Run playbooks in an Ansible controller node
+Each Ansible controller node contains an Ansible inventory together with all different playbooks and files used together with the playbooks. The targets for the different tasks of a playbook are defined inside the playbook yaml file itself and not externally like Salt does for the states.
+
+This is an example of Ansible playbook:
+
+```yaml
+---
+- name: update web servers
+  hosts: webservers
+  remote_user: root
+
+  tasks:
+  - name: ensure apache is at the latest version
+    yum:
+      name: httpd
+      state: latest
+  - name: write the apache config file
+    template:
+      src: /srv/httpd.j2
+      dest: /etc/httpd.conf
+
+- name: update db servers
+  hosts: databases
+  remote_user: root
+
+  tasks:
+  - name: ensure postgresql is at the latest version
+    yum:
+      name: postgresql
+      state: latest
+  - name: ensure that postgresql is started
+    service:
+      name: postgresql
+      state: started
+```
+
+As you can see, this playbook is defining different tasks to execute on different "hosts" groups. Those groups are of course defined in the Ansible inventory (default at /etc/ansible/hosts).
+
+The execution of a given playbooks is done in the corresponding Ansible controller, which contains the inventory and the hosts definition.
+
+The playbooks available to apply on each Ansible controller, are the ones exposed by "ansible-gatherer" and ultimately, in the future, playbooks created and maintained in Uyuni (via future Ansible playbook catalog) that would be then pushed to the controller node.
+
+At the time of running a playbook from the Ansible controller, Uyuni is able to do it by reaching the controller and operating it using "salt-ssh". There are different approaches here:
```

Comment:
Why is operating Ansible the Ansible Controller node limited to `salt-ssh`? In other words, why is `salt my-ansible-controller ...` not an alternative?

Reply:
Thanks for the feedback @agraul ! :+1:

I've added few notes that I think clarify this part of the text a little bit.

I mentioned using "salt-ssh" because at this point, it was not required that the ansible controller is a registered system, so in this case, Uyuni would use the "ssh" credentials provided to "ansible-gatherer" in order to reach the controller.

Of course, if the Ansible controller is a registered system, we do not need to do a Salt SSH, simply trigger the playbook execution via normal Salt minion job.

Reply:
@meaksh just to clarify: we are considering the case where the Uyuni Server would be the Ansible controller, right? Somehow we need to make this case easy to set up (especially considering the Uyuni Server cannot be managed by itself, so entitlements or formulas are not an option).

Reply:
@paususe yes, I've added some notes here.